### PR TITLE
REST API uploads: fix year/month logic

### PR DESCRIPTION
**Pull request by swissspidy**
Trac ticket: https://core.trac.wordpress.org/ticket/61189

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by adamsilverstein — APPROVED** [review]
Nice!

File: src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php [hunk 1]
```diff
@@ -1035,12 +1046,14 @@ public function get_item_schema() {
...
 	 * @since 4.7.0
+	 * @since 6.6.0 Added the `$time` parameter.
 	 *
 	 * @param string $data    Supplied file data.
 	 * @param array  $headers HTTP headers from the request.
+	 * @param string $time    Optional. Time formatted in 'yyyy/mm'. Default null.
 	 * @return array|WP_Error Data from wp_handle_sideload().
 	 */
-	protected function upload_from_data( $data, $headers ) {
+	protected function upload_from_data( $data, $headers, $time ) {
```
Comment (TimothyBJacobs) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php]:
```suggestion
	protected function upload_from_data( $data, $headers, $time = null ) {
```

**Comment by TimothyBJacobs** [comment]
nice fix!

File: src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php (hunk 1 above)
Comment (swissspidy) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php]: Good catch!

File: src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php [hunk 2]
```diff
@@ -1246,12 +1259,14 @@ public function get_collection_params() {
 	 * Handles an upload via multipart/form-data ($_FILES).
 	 *
 	 * @since 4.7.0
+	 * @since 6.6.0 Added the `$time` parameter.
 	 *
-	 * @param array $files   Data from the `$_FILES` superglobal.
-	 * @param array $headers HTTP headers from the request.
+	 * @param array  $files   Data from the `$_FILES` superglobal.
+	 * @param array  $headers HTTP headers from the request.
+	 * @param string $time    Optional. Time formatted in 'yyyy/mm'. Default null.
```
Comment (todeveni) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php]: `@param string|null $time` if it's nullable (and `null` by default)

Comment (swissspidy) [reply src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php]: Good point. I should fix this in the places where I copied this from

**Review by SergeyBiryukov — APPROVED** [review]
Looks good to me :+1:

**Comment by swissspidy** [comment]
Committed in https://core.trac.wordpress.org/changeset/58130

**Pull request closed without merging**